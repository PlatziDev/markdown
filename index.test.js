const createParser = require('./index.js');
const math = require('markdown-it-math');

const markdown = `![Platzi Logo](https://static.platzi.com/static/images/logos/platzi@2x.png)

*[HTML]: Hyper Text Markup Language
*[W3C]:  World Wide Web Consortium

# Heading 1
Lorem **ipsum** dolor _sit_ ++amet++ :D.

The HTML ==specification== is maintained by the W3C.

- \`item 1\`
- item 2
- item 3

Term 1
  ~ Definition 1

Term 2
  ~ Definition 2a
  ~ Definition 2b

\`\`\`js
function greeting() {
	return 'hello world!';
}
\`\`\`

@S[soundcloud](https://soundcloud.com/platziteam/la-historia-de-platzi)

@[youtube](ajLJOhf-WdA)`;

const markdownWithMath = `Pythagoran theorem is $$a^2 + b^2 = c^2$$.

Bayes theorem:

$$$
P(A | B) = (P(B | A)P(A)) / P(B)
$$$`;

describe('Platzi Flavored Markdown parser', () => {
  it('should work without arguments', () => {
    expect(createParser()(markdown)).toMatchSnapshot();
  });

  describe('Options test', () => {
    it('should work with options', () => {
      expect(
        createParser({
          html: false
        })(markdown)
      ).toMatchSnapshot();
    });

    it('should break without an object as options', () => {
      expect(() => createParser('fake options')).toThrowErrorMatchingSnapshot();
    });
  });

  describe('Extra plugins test', () => {
    it('should work with extra plugins', () => {
      expect(
        createParser(undefined, [
          math,
          [
            math,
            {
              inlineOpen: '$$'
            }
          ]
        ])(markdownWithMath)
      ).toMatchSnapshot();
    });

    it('should break without an array as extra plugins', () => {
      expect(() =>
        createParser(undefined, 'fake extra plugins')
      ).toThrowErrorMatchingSnapshot();
    });
  });
});
