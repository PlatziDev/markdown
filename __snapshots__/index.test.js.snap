// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Platzi Flavored Markdown parser Extra plugins test should break without an array as extra plugins 1`] = `"The parser extra plugins must be an array."`;

exports[`Platzi Flavored Markdown parser Extra plugins test should work with extra plugins 1`] = `
"<p>Pythagoran theorem is <math><msup><mi>a</mi><mn>2</mn></msup><mo>+</mo><msup><mi>b</mi><mn>2</mn></msup><mo>=</mo><msup><mi>c</mi><mn>2</mn></msup></math>.</p>
<p>Bayes theorem:</p>
<math display=\\"block\\"><mi>P</mi><mfenced open=\\"(\\" close=\\")\\"><mrow><mi>A</mi><mo stretchy=\\"true\\" lspace=\\"veryverythickmathspace\\" rspace=\\"veryverythickmathspace\\">|</mo><mi>B</mi></mrow></mfenced><mo>=</mo><mfrac><mrow><mi>P</mi><mfenced open=\\"(\\" close=\\")\\"><mrow><mi>B</mi><mo stretchy=\\"true\\" lspace=\\"veryverythickmathspace\\" rspace=\\"veryverythickmathspace\\">|</mo><mi>A</mi></mrow></mfenced><mi>P</mi><mfenced open=\\"(\\" close=\\")\\"><mi>A</mi></mfenced></mrow><mrow><mi>P</mi><mfenced open=\\"(\\" close=\\")\\"><mi>B</mi></mfenced></mrow></mfrac></math>
"
`;

exports[`Platzi Flavored Markdown parser Options test should break without an object as options 1`] = `"The markdown parser options must be an object."`;

exports[`Platzi Flavored Markdown parser Options test should work with options 1`] = `
"<figure><img src=\\"https://static.platzi.com/static/images/logos/platzi@2x.png\\" alt=\\"Platzi Logo\\" /></figure>
<h1>Heading 1</h1>
<p>Lorem <strong>ipsum</strong> dolor <em>sit</em> <ins>amet</ins> ðŸ˜„.</p>
<p>The <abbr title=\\"Hyper Text Markup Language\\">HTML</abbr> <mark>specification</mark> is maintained by the <abbr title=\\"World Wide Web Consortium\\">W3C</abbr>.</p>
<ul>
<li><code>item 1</code></li>
<li>item 2</li>
<li>item 3</li>
</ul>
<dl>
<dt>Term 1</dt>
<dd>Definition 1</dd>
<dt>Term 2</dt>
<dd>Definition 2a</dd>
<dd>Definition 2b</dd>
</dl>
<pre><code class=\\"language-js\\">function greeting() {
	return 'hello world!';
}
</code></pre>
<p><div><iframe width=\\"100%\\" height=\\"450\\" scrolling=\\"no\\" frameborder=\\"no\\" src=\\"https://w.soundcloud.com/player/?url=https://soundcloud.com/platziteam/la-historia-de-platzi&amp;auto_play=false&amp;hide_related=false&amp;show_comments=true&amp;show_user=true&amp;show_reposts=false&amp;visual=true\\"></iframe></div></p>
<p><div class=\\"embed-responsive embed-responsive-16by9\\"><iframe class=\\"embed-responsive-item youtube-player\\" type=\\"text/html\\" width=\\"640\\" height=\\"390\\" src=\\"https://www.youtube.com/embed/ajLJOhf-WdA\\" frameborder=\\"0\\" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe></div></p>
<p><a href=\\"http://platzi.com/\\" target=\\"_blank\\" rel=\\"nofollow noopener\\">platzi</a></p>
"
`;

exports[`Platzi Flavored Markdown parser should work without arguments 1`] = `
"<figure><img src=\\"https://static.platzi.com/static/images/logos/platzi@2x.png\\" alt=\\"Platzi Logo\\" /></figure>
<h1>Heading 1</h1>
<p>Lorem <strong>ipsum</strong> dolor <em>sit</em> <ins>amet</ins> ðŸ˜„.</p>
<p>The <abbr title=\\"Hyper Text Markup Language\\">HTML</abbr> <mark>specification</mark> is maintained by the <abbr title=\\"World Wide Web Consortium\\">W3C</abbr>.</p>
<ul>
<li><code>item 1</code></li>
<li>item 2</li>
<li>item 3</li>
</ul>
<dl>
<dt>Term 1</dt>
<dd>Definition 1</dd>
<dt>Term 2</dt>
<dd>Definition 2a</dd>
<dd>Definition 2b</dd>
</dl>
<pre><code class=\\"language-js\\">function greeting() {
	return 'hello world!';
}
</code></pre>
<p><div><iframe width=\\"100%\\" height=\\"450\\" scrolling=\\"no\\" frameborder=\\"no\\" src=\\"https://w.soundcloud.com/player/?url=https://soundcloud.com/platziteam/la-historia-de-platzi&amp;auto_play=false&amp;hide_related=false&amp;show_comments=true&amp;show_user=true&amp;show_reposts=false&amp;visual=true\\"></iframe></div></p>
<p><div class=\\"embed-responsive embed-responsive-16by9\\"><iframe class=\\"embed-responsive-item youtube-player\\" type=\\"text/html\\" width=\\"640\\" height=\\"390\\" src=\\"https://www.youtube.com/embed/ajLJOhf-WdA\\" frameborder=\\"0\\" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe></div></p>
<p><a href=\\"http://platzi.com/\\" target=\\"_blank\\" rel=\\"nofollow noopener\\">platzi</a></p>
"
`;
